src/main/java/com/example/store/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.example.store.service.impl;
+
+import com.example.store.dto.cartitem.CartItemDto;
+import com.example.store.dto.cartitem.CartItemRequestDto;
+import com.example.store.dto.cartitem.CartItemUpdateRequestDto;
+import com.example.store.exception.EntityNotFoundException;
+import com.example.store.exception.UnauthorizedModificationException;
+import com.example.store.mapper.CartItemMapper;
+import com.example.store.model.CartItem;
+import com.example.store.model.ShoppingCart;
+import com.example.store.repository.book.BookRepository;
+import com.example.store.repository.cartitem.CartItemRepository;
+import com.example.store.service.CartItemService;
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public CartItem save(CartItemRequestDto requestDto, ShoppingCart shoppingCart) {
```

Comment:
```suggestion
    public CartItem addOrUpdateCartItem(CartItemRequestDto requestDto, ShoppingCart shoppingCart) {
```
maybe smth like this to better reflect it's logic